### ISSUE TITLE
`<label>` doesn't keep `htmlFor` attribute between renders

### ISSUE THREAD

**Issue by mglikesbikes** [BuilderIO/qwik#1384]
### Qwik Version

0.9.0

### Operating System (or Browser)

--

### Which component is affected?

Qwik Runtime

### Expected Behaviour

Seeing a `<label>` lose its `for` attributes between renders. [Here's a Playground](https://qwik.builder.io/playground#version=0.9.0&buildMode=development&entryStrategy=hook&files=JTVCJTdCJTIycGF0aCUyMiUzQSUyMiUyRmFwcC50c3glMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJTdCJTIwY29tcG9uZW50JTI0JTJDJTIwdXNlU3RvcmUlMjAlN0QlMjBmcm9tJTIwJyU0MGJ1aWxkZXIuaW8lMkZxd2lrJyUzQiU1Q24lNUNudHlwZSUyMEFuaW1hbCUyMCUzRCUyMCU1QyUyMmRvZyU1QyUyMiU3QyU1QyUyMmNhdCU1QyUyMiU3QyU1QyUyMiU1QyUyMiU1Q24lNUNuZXhwb3J0JTIwY29uc3QlMjBBcHAlMjAlM0QlMjBjb21wb25lbnQlMjQoKCklMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwY29uc3QlMjBzdG9yZSUyMCUzRCUyMHVzZVN0b3JlJTNDJTdCX2ZpZWxkSWQlM0FzdHJpbmclMkNhbmltYWwlM0FBbmltYWwlN0QlM0UoJTdCJTVDbiUyMCUyMCUyMCUyMF9maWVsZElkJTNBJTIwJTVDJTIyZm9vJTVDJTIyJTJDJTVDbiUyMCUyMCUyMCUyMGFuaW1hbCUzQSUyMCcnJTVDbiUyMCUyMCU3RCklM0IlNUNuJTVDbiUyMCUyMHJldHVybiUyMCglNUNuJTIwJTIwJTIwJTIwJTNDJTNFJTVDbiUyMCUyMCUyMCUyMCUzQ2ZpZWxkc2V0JTIwc3R5bGUlM0QlN0IlN0JkaXNwbGF5JTNBJTIwJ2ZsZXgnJTJDJTIwJ2ZsZXgtZGlyZWN0aW9uJyUzQSUyMCdjb2x1bW4nJTdEJTdEJTNFJTVDbiUyMCUyMCUyMCUyMCUzQ2gzJTNFUmVwcm9kdWN0aW9uJTNDJTJGaDMlM0UlNUNuJTIwJTIwJTIwJTIwJTNDcCUzRUlzc3VlJTNBJTIwbGFiZWwlMjBsb3NlcyUyMHRoZSUyMCU2MGZvciU2MCUyMGF0dHJpYnV0ZSUyMGJldHdlZW4lMjByZW5kZXJzLiUzQyUyRnAlM0UlNUNuJTIwJTIwJTIwJTIwJTNDb2wlMjBzdHlsZSUzRCU3QiU3Qm1hcmdpbiUzQSUyMCcwJTIwMCUyMDFyZW0nJTdEJTdEJTNFJTVDbiUyMCUyMCUyMCUyMCUzQ2xpJTNFUmlnaHQlMjBjbGljayUyMHRoZSUyMGxhYmVsJTIwJTI2cmFyciUzQiUyMGluc3BlY3QlMjBlbGVtZW50JTNDJTJGbGklM0UlNUNuJTIwJTIwJTIwJTIwJTNDbGklM0VDaGFuZ2UlMjB0aGUlMjB2YWx1ZSUyMG9mJTIwdGhlJTIwc2VsZWN0JTNDJTJGbGklM0UlNUNuJTIwJTIwJTIwJTIwJTNDbGklM0VWaWElMjBEZXZUb29scyUyQyUyMG5vdGUlMjB0aGUlMjBsYWJlbCdzJTIwbWlzc2luZyUyMGElMjAlNjBmb3IlNjAlMjBhdHRyaWJ1dGUlM0MlMkZsaSUzRSU1Q24lMjAlMjAlMjAlMjAlM0MlMkZvbCUzRSU1Q24lNUNuJTIwJTIwJTIwJTIwJTdCJTJGKiUyMCUyMCUyMCUyMCVFMiVBQyU4NyVFRiVCOCU4RiUyMHRoaXMlMjBhdHRyaWJ1dGUlMjBkaXNhcHBlYXJzJUUyJTgwJUE2KiUyRiU3RCU1Q24lMjAlMjAlMjAlMjAlM0NsYWJlbCUyMGh0bWxGb3IlM0QlN0JzdG9yZS5fZmllbGRJZCU3RCUzRUknbSUyMGElMjBsYWJlbCUzQSUzQyUyRmxhYmVsJTNFJTVDbiUyMCUyMCUyMCUyMCU3QiUyRiolMjAlMjAlMjAlMjAlRTIlQUMlODclRUYlQjglOEYlMjB3aGVuJTIwdGhpcyUyMCUzQ3NlbGVjdCUzRSUyMGlzJTIwY2hhbmdlZColMkYlN0QlNUNuJTIwJTIwJTIwJTIwJTNDc2VsZWN0JTIwaWQlM0QlN0JzdG9yZS5fZmllbGRJZCU3RCUyMHZhbHVlJTNEJTdCc3RvcmUuYW5pbWFsJTdEJTIwb25DaGFuZ2UlMjQlM0QlN0IoZSUzQUV2ZW50KSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjBzdG9yZS5hbmltYWwlMjAlM0QlMjAoZS50YXJnZXQlMjBhcyUyMEhUTUxTZWxlY3RFbGVtZW50KS52YWx1ZSUyMGFzJTIwQW5pbWFsJTNCJTVDbiUyMCUyMCUyMCUyMCU3RCU3RCUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlM0NvcHRpb24lMjB2YWx1ZSUzRCU1QyUyMiU1QyUyMiUyMHNlbGVjdGVkJTNEJTdCc3RvcmUuYW5pbWFsJTIwJTNEJTNEJTNEJTIwJyclN0QlMjBkaXNhYmxlZCUzRVBpY2slMjBvbmUlM0MlMkZvcHRpb24lM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTNDb3B0aW9uJTIwdmFsdWUlM0QlNUMlMjJkb2clNUMlMjIlM0VEb2clM0MlMkZvcHRpb24lM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTNDb3B0aW9uJTIwdmFsdWUlM0QlNUMlMjJjYXQlNUMlMjIlM0VDYXQlM0MlMkZvcHRpb24lM0UlNUNuJTIwJTIwJTIwJTIwJTNDJTJGc2VsZWN0JTNFJTVDbiU1Q24lMjAlMjAlMjAlMjAlN0JzdG9yZS5hbmltYWwlMjAlMjYlMjYlMjAoJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUzQyUzRVNlbGVjdGVkJTNBJTIwJTdCc3RvcmUuYW5pbWFsJTdEJTNDJTJGJTNFJTVDbiUyMCUyMCUyMCUyMCklN0QlNUNuJTIwJTIwJTIwJTIwJTNDJTJGZmllbGRzZXQlM0UlNUNuJTIwJTIwJTIwJTIwJTNDJTJGJTNFJTVDbiUyMCUyMCklM0IlNUNuJTdEKSUzQiU1Q24lMjIlN0QlMkMlN0IlMjJwYXRoJTIyJTNBJTIyJTJGZW50cnkuc2VydmVyLnRzeCUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAlN0IlMjByZW5kZXJUb1N0cmluZyUyQyUyMFJlbmRlck9wdGlvbnMlMjAlN0QlMjBmcm9tJTIwJyU0MGJ1aWxkZXIuaW8lMkZxd2lrJTJGc2VydmVyJyUzQiU1Q25pbXBvcnQlMjAlN0IlMjBSb290JTIwJTdEJTIwZnJvbSUyMCcuJTJGcm9vdCclM0IlNUNuJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjBmdW5jdGlvbiUyMChvcHRzJTNBJTIwUmVuZGVyT3B0aW9ucyklMjAlN0IlNUNuJTIwJTIwcmV0dXJuJTIwcmVuZGVyVG9TdHJpbmcoJTNDUm9vdCUyMCUyRiUzRSUyQyUyMG9wdHMpJTNCJTVDbiU3RCU1Q24lMjIlN0QlMkMlN0IlMjJwYXRoJTIyJTNBJTIyJTJGcm9vdC50c3glMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJTdCJTIwQXBwJTIwJTdEJTIwZnJvbSUyMCcuJTJGYXBwJyUzQiU1Q24lNUNuZXhwb3J0JTIwY29uc3QlMjBSb290JTIwJTNEJTIwKCklMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwcmV0dXJuJTIwKCU1Q24lMjAlMjAlMjAlMjAlM0NodG1sJTNFJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUzQ2hlYWQlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTNDdGl0bGUlM0VIZWxsbyUyMFF3aWslM0MlMkZ0aXRsZSUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlM0MlMkZoZWFkJTNFJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUzQ2JvZHklM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTNDQXBwJTIwJTJGJTNFJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUzQyUyRmJvZHklM0UlNUNuJTIwJTIwJTIwJTIwJTNDJTJGaHRtbCUzRSU1Q24lMjAlMjApJTNCJTVDbiU3RCUzQiU1Q24lMjIlN0QlNUQ%3D) demo'ing the issue. 

```
import { component$, useStore } from '@builder.io/qwik';

type Animal = "dog"|"cat"|""

export const App = component$(() => {
  const store = useStore<{_fieldId:string,animal:Animal}>({
    _fieldId: "foo",
    animal: ''
  });

  return (
    <>
    <fieldset style={{display: 'flex', 'flex-direction': 'column'}}>
    <h3>Reproduction</h3>
    <p>Issue: label loses the `for` attribute between renders.</p>
    <ol style={{margin: '0 0 1rem'}}>
    <li>Right click the label &rarr; inspect element</li>
    <li>Change the value of the select</li>
    <li>Via DevTools, note the label's missing a `for` attribute</li>
    </ol>

    {/*    ⬇️ this attribute disappears…          */}
    <label htmlFor={store._fieldId}>I'm a label:</label>
    {/*    ⬇️ when this <select> is changed.  */}
    <select id={store._fieldId} value={store.animal} onChange$={(e:Event) => {
      store.animal = (e.target as HTMLSelectElement).value as Animal;
    }}>
      <option value="" selected={store.animal === ''} disabled>Pick one</option>
      <option value="dog">Dog</option>
      <option value="cat">Cat</option>
    </select>

    {store.animal && (
      <>Selected: {store.animal}</>
    )}
    </fieldset>
    </>
  );
});
```

### Actual Behaviour

See above

**Comment by imhappyfor** [comment]
Interesting issue, I did a dive into the code but fell short of any answers. For others trying to fix this bug, adding this store change for the "private" _fieldId will give the same output as above on the first attribute change after the initial render. Subsequent updates to the select field will produce the desired behavior of setting the for attribute correctly.

```
    <select id={store._fieldId} value={store.animal} onChange$={(e:Event) => {
    **store._fieldId = (e.target as HTMLSelectElement).value as Animal;**
      store.animal = (e.target as HTMLSelectElement).value as Animal;
    }}>
```

**Comment by manucorporat** [comment]
This should be fixed in latest 0.10.0, use the for attribute instead!
